### PR TITLE
feat: incident reporting for providers (closes #34)

### LINKED ISSUE #34: [FEATURE]  Incident Reporting System for Providers
**Feature Description**
Providers need a way to formally document incidents, safety concerns, or noteworthy events that occur during programs. This protects both providers and the platform while creating an audit trail for serious issues. 
Build an incident reporting system that:

Allows providers to submit timestamped incident reports
Includes categories (safety concern, behavioral issue, injury, property damage, etc.)

Creates a secure, archived record accessible to authorized personnel

**User Story**
As a instructor], I want report incidents so that i have a log just in case its needed for disputes or reference

**Acceptance Criteria**
 [] Incident report button is prominently accessible in provider dashboard
 [] Report form captures: date/time, program/session ID, incident category, severity level
 [] Predefined categories available: Safety Concern, Behavioral Issue, Injury, Property Damage, Policy Violation, Other
 [] Severity levels: Low, Medium, High, Critical
[] For Prime Youth Providers, Sends immediate notifications to platform administrators for urgent issues
[] Supports photo/video evidence upload

**Additional context**
Any other context, mockups, or examples about the feature.

### PR DESCRIPTION
**Pull request by MaxPayne89**
## Summary

Implements end-to-end incident reporting for providers (issue #34): a dedicated `/provider/incidents/new` page accessible from the dashboard's Program Inventory action row, with photo upload, polymorphic program-or-session scope, and event-driven publishing of `incident_reported` domain + integration events. Also introduces a Provider-local `provider_programs` projection fed by Program Catalog events (#735 will generalize the projection pattern).

## Architecture

- **Domain model:** `KlassHero.Provider.Domain.Models.IncidentReport` — pure struct with `new/1` enforcing 6 invariants (XOR program/session, category/severity enums, description min length, occurred_at not in future, photo URL/filename pairing).
- **Persistence:** Standard schema/mapper/port/repository stack. Migration includes a polymorphic CHECK constraint and category/severity CHECKs.
- **Cross-context data via projection (no ACLs):** new `ProviderPrograms` GenServer subscribes to `integration:program_catalog:program_created` and `:program_updated` topics, maintains the `provider_programs` read table with bootstrap-on-init + retry-with-backoff. The `SubmitIncidentReport` command validates program ownership locally via this projection — no synchronous cross-context calls.
- **Event-driven:** Command publishes `:incident_reported` domain event via `DomainEventBus.dispatch/2`. The existing `PromoteIntegrationEvents` handler escalates it to a critical (Oban-backed) integration event with payload containing only metadata — no `description` or `photo_url` (intentionally — events must not leak sensitive content).
- **Web layer:** Dedicated `IncidentReportLive` page mirrors `profile_completion_live.ex` patterns (Layouts.app via live_session, `<.input>` components, drag-drop upload via `live_file_input`). Boundary-clean: LiveView calls `Provider.submit_incident_report/1` facade.

## Review Focus

1. **`ProviderPrograms` projection** (`lib/klass_hero/provider/adapters/driven/projections/provider_programs.ex`) — new GenServer with a `@default_status \"active\"` constant because `ProgramSchema` has no status field today. Tracked for cleanup in #736.
2. **`SubmitIncidentReport` command** (`lib/klass_hero/provider/application/commands/incident/submit_incident_report.ex`) — `with` pipeline orchestrates ownership validation → upload → persist → event. `:nilify_all` was rejected for `reporter_user_id` FK in favor of `:restrict` (matches `messages.sender_id` precedent).
3. **Event payload shape** (`KlassHero.Provider.Domain.Events.ProviderEvents.incident_reported/2`) — explicitly omits `description` and `photo_url`. Tests assert via `refute Map.has_key?`. Future consumers (notifications, admin queue) must fetch fresh data via a query.
4. **Polymorphic invariant defended at three layers** — domain model `validate_target/1`, Ecto changeset `check_constraint(:target, ...)`, DB-level `CHECK ((program_id IS NOT NULL) <> (session_id IS NOT NULL))`.

## Test Plan

- [x] **Unit:** `IncidentReport` domain model — 14 tests covering all 6 invariants + accessors + edge cases (unknown keys ignored, multi-byte description handling)
- [x] **Unit:** Domain & integration event factories — 6 tests covering payload shape + sensitive-field exclusion + opts passthrough
- [x] **Persistence:** `IncidentReportRepository.create/1` — 2 tests for happy path + FK error
- [x] **Projection:** `ProviderPrograms` GenServer — 3 tests for `:program_created` upsert, `:program_updated` update, `rebuild/1` reprojection from write table
- [x] **Repository:** `ProviderProgramRepository` — 4 tests for `get_by_id/1` + `list_by_provider/1`
- [x] **Port extension:** `SessionDetailsRepository.get_by_id/1` — 2 tests
- [x] **Application:** `SubmitIncidentReport.execute/1` — 5 tests (program-scope happy path, session-scope happy path, photo-upload happy path, ownership-fail, validation-fail)
- [x] **LiveView:** `IncidentReportLive` — 7 tests (mount, preselection, redirect-on-invalid, all form fields render, photo dropzone, submit happy path, submit error path)
- [x] **Component:** Program Inventory action button — 1 test asserting the link target
- [x] **Integration:** `mix precommit` clean — 4533 passed, 12 skipped, 0 failures

## Follow-up issues filed

- #735 — Generalize projection boilerplate via macro/DSL
- #736 — Drop speculative `provider_programs.status` column
- #737 — Incident report lifecycle (under_review/resolved)
- #738 — Provider \"My Incidents\" list page
- #739 — Admin incident queue LiveView
- #740 — Multi-photo attachments
- #741 — Refactor dashboard to read programs from projection
- #742 — Notification workflow consuming incident_reported event

Closes #34

### CODE REVIEW

File: lib/klass_hero/provider/application/commands/incident/submit_incident_report.ex [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  defp maybe_upload_photo(%{file_binary: file_binary} = params) when is_binary(file_binary) do
+    path = build_path(params.provider_profile_id, params[:original_filename])
+    content_type = params[:content_type] || "image/jpeg"
+    opts = Keyword.merge([content_type: content_type], params[:storage_opts] || [])
+
+    with {:ok, key} <- Storage.upload(:private, path, file_binary, opts) do
+      {:ok, %{photo_url: key, original_filename: params[:original_filename]}}
+    end
+  end
+
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero/provider/application/commands/incident/submit_incident_report.ex]: Addressed in e7234385 — `maybe_upload_photo/1` now validates that `original_filename` is a non-blank binary before calling `Storage.upload/4`. New clause returns `{:error, [original_filename: "is required when photo is uploaded"]}` for the "binary without filename" case. Added a regression test that asserts no upload happens when the filename is missing.

File: priv/repo/migrations/20260423214232_create_incident_reports.exs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      add :id, :binary_id, primary_key: true
+
+      add :provider_id, references(:providers, type: :binary_id, on_delete: :delete_all),
+        null: false
+
+      add :reporter_user_id, references(:users, type: :binary_id, on_delete: :restrict),
+        null: false
+
+      add :program_id, references(:programs, type: :binary_id, on_delete: :nilify_all)
+      add :session_id, references(:program_sessions, type: :binary_id, on_delete: :nilify_all)
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply priv/repo/migrations/20260423214232_create_incident_reports.exs]: Addressed in e7234385 — both `program_id` and `session_id` FKs now use `on_delete: :restrict`, matching the policy already applied to `reporter_user_id`. The XOR CHECK still catches the same scenarios, but `:restrict` makes the intent explicit and produces a clearer error than a deferred constraint violation. Verified the live DB now shows `ON DELETE RESTRICT` for both FKs.

File: lib/klass_hero_web/live/provider/incident_report_live.ex [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+          />
+
+          <.input
+            field={@form[:program_id]}
+            type="select"
+            label={gettext("Program")}
+            options={Enum.map(@programs, &{&1.name, &1.program_id})}
+            prompt={gettext("Select a program")}
+            required
+          />
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero_web/live/provider/incident_report_live.ex:279]: Addressed in e7234385 — removed the half-implemented session pathway and aligned the moduledoc with the actual behavior. The use case still supports session-scoped reports; the UI surface for them is tracked in #738.

File: lib/klass_hero_web/live/provider/incident_report_live.ex [hunk 4]
```diff
@@ -0,0 +1,363 @@
...
+    case @programs_query.get_by_id(program_id) do
+      {:ok, %{provider_id: ^provider_id} = program} -> {:ok, {:program, program}}
+      _ -> {:error, :preselection_invalid}
+    end
+  end
+
+  defp resolve_preselection(%{"session_id" => session_id}, _provider_id) when is_binary(session_id) do
+    # Session-scope preselection wiring lands in a future task.
+    {:ok, :none}
+  end
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero_web/live/provider/incident_report_live.ex]: Addressed in e7234385 — deleted the dead `resolve_preselection(%{"session_id" => _}, _)` clause. `?session_id=` now naturally falls through to the `:none` catch-all, matching the documented behavior. Future session-preselection work tracked in #738.

File: lib/klass_hero/provider/adapters/driven/projections/provider_programs.ex [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+  Programs in the Program Catalog have no first-class status field today, so
+  the projection records every program as `"active"`. This column exists to
+  support future filtering (e.g. archived/draft) without requiring a migration.
+  """
+
+  use GenServer
+
+  import Ecto.Query
+
+  alias KlassHero.ProgramCatalog.Adapters.Driven.Persistence.Schemas.ProgramSchema
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero/provider/adapters/driven/projections/provider_programs.ex:41]: Acknowledged but not addressing in this PR. This is the same projection-bootstrap pattern used by `KlassHero.ProgramCatalog.Adapters.Driven.Projections.ProgramListings`, which reads from its own context's write table at cold start. Both reach across context boundaries only inside `bootstrap_from_write_table/0` (cold start + explicit `rebuild/1`); runtime is purely event-driven via `integration:program_catalog:*` PubSub topics, so there is no synchronous cross-context read on the hot path.

The `boundary` library is satisfied (CI compile is clean) and `mix precommit` passes. Since the same shape exists in `ProgramListings` already on `main`, fixing it in isolation here would create inconsistency. Tracked in #735, which will introduce a projection macro that encapsulates "bootstrap from upstream context" as a first-class concept and eliminates the direct schema reach in both projections together.

File: lib/klass_hero_web/live/provider/incident_report_live.ex [hunk 6]
```diff
@@ -0,0 +1,363 @@
...
+    end
+  end
+
+  defp safe_consume_photo(socket) do
+    {:ok,
+     consume_uploaded_entries(socket, :photo, fn %{path: path}, entry ->
+       # sobelow_skip ["Traversal.FileModule"]
+       {:ok,
+        %{
+          binary: File.read!(path),
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero_web/live/provider/incident_report_live.ex:156]: False positive — the `path` argument comes from `Phoenix.LiveView.consume_uploaded_entries/3`, which gives us a temporary path that Phoenix itself wrote to and manages. It is not user-supplied input. The same pattern is already in production at `profile_completion_live.ex:148` (verification document upload), and the project's `.sobelow-conf` does not skip `Traversal.FileModule`. CI uses `exit: :high`, so these MEDIUM warnings don't fail the build. Dismissing per the established codebase pattern; if we want zero-warning hygiene, that would be a separate cleanup spanning all upload sites.

File: lib/klass_hero_web/live/provider/incident_report_live.ex [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero_web/live/provider/incident_report_live.ex:156]: Same finding as comment 3135579842 — false positive. The `path` comes from Phoenix-managed upload temp directory, not user input. The new commit also adds an explicit `# sobelow_skip [\"Traversal.FileModule\"]` annotation at the call site to suppress this in future scanner runs.

File: lib/klass_hero_web/live/provider/incident_report_live.ex [hunk 8]
```diff
@@ -0,0 +1,357 @@
...
```
[the remark this answers is not in the document]
Comment (MaxPayne89) [reply lib/klass_hero_web/live/provider/incident_report_live.ex:156]: Same finding as comment 3135579842 — false positive. The `path` comes from Phoenix-managed upload temp directory, not user input. The new commit also adds an explicit `# sobelow_skip [\"Traversal.FileModule\"]` annotation at the call site to suppress this in future scanner runs.
